cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -175,9 +177,33 @@ func (p *Planner) injectOngoingActuation() error {
 	if err != nil {
 		return err
 	}
-	// TODO(x13n): Check owner references to avoid double-counting already
-	// recreated pods.
-	return p.injectPods(p.actuationStatus.RecentEvictions())
+	return p.injectPods(filterOutRecreatedPods(p.actuationStatus.RecentEvictions(), p.cc))
+}
+
+func filterOutRecreatedPods(pods []*apiv1.Pod, cc ControllerReplicasCalculator) []*apiv1.Pod {
+	var podsToInject []*apiv1.Pod
+	addedReplicas := make(map[string]int32)
+	for _, pod := range pods {
+		if len(pod.GetOwnerReferences()) == 0 {
+			podsToInject = append(podsToInject, pod)
+			continue
+		}
+		ownerRef := pod.GetOwnerReferences()[0]
+		if ownerRef.Kind == "DaemonSet" {
```

Comment:
nit: I don't think DaemonSet really requires special handling here, `GetReplicas` can just return current=target. Might feel a bit hacky though.

Reply:
I feel like it's a little too hacky to not handle DaemonSet here